app/components/address-register-selector.js

```diff
@@ -0,0 +1,63 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+
+export default class AddressRegisterSelectorComponent extends Component {
+  @service addressRegister;
+  @service store;
+
+  @tracked addressSuggestion;
+
+  sourceCrab;
+
+  get isDisabledBusNumberSelect() {
+    return !this.addressWithBusNumber;
+  }
+
+  constructor() {
+    super(...arguments);
+    if (this.args.address) {
+      this.addressSuggestion = this.addressRegister.toAddressSuggestion(
+        this.args.address
+      );
+    }
+  }
+
+  @task
+  *selectSuggestion(addressSuggestion) {
+    this.addressSuggestion = addressSuggestion;
+
+    if (addressSuggestion) {
+      const addresses = yield this.addressRegister.findAll(addressSuggestion);
+      if (!this.sourceCrab) {
+        this.sourceCrab = yield this.store.findRecord(
+          'concept',
+          'e59c97a9-4e95-4d65-9696-756de47fbc1f'
+        );
+      }
+      // TODO: this should probably be fixed in the API itself (, if possible)
```

Comment:
it's crab itself i think that has duplicates (uri are differents if i remember, but address seems to be exactly the same)